src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -210,16 +212,27 @@ private static ProcessModuleCollection GetModules(int processId, bool firstModul
 
                         module.ModuleName = new string(chars, 0, length);
 
-                        length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                        for (; ; )
+                        {
+                            length = Interop.Kernel32.GetModuleFileNameEx(processHandle, moduleHandle, chars, chars.Length);
+                            if (length == chars.Length)
+                            {
+                                ArrayPool<char>.Shared.Return(chars);
+                                minimumLength *= 2;
+                                chars = ArrayPool<char>.Shared.Rent(minimumLength);
+                                continue;
+                            }
+
+                            break;
+                        }
+
                         if (length == 0)
                         {
                             HandleError();
                             continue;
                         }
 
-                        module.FileName = (length >= 4 && chars[0] == '\\' && chars[1] == '\\' && chars[2] == '?' && chars[3] == '\\') ?
```

Comment:
Why is this logic getting removed?

Reply:
I don't think the "\\\\?\\" prefix should be stripped. Most APIs (e.g. `Path.GetFullPath`) return paths with this prefix.

https://github.com/dotnet/corefx/pull/21746#discussion_r125149958 /cc @JeremyKuhne

Reply:
This PR is primarily style cleanup, with a few perf improvements.  Little functional changes like this one tucked in make it very difficult to review.  Please separate it out, at which point we can have a conversation about it.